builtin/providers/aws/resource_aws_internet_gateway.go

```diff
@@ -50,6 +50,27 @@ func resourceAwsInternetGatewayCreate(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
+	resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DescribeInternetGateways(&ec2.DescribeInternetGatewaysInput{
```

Comment:
The response here should be able to tell us something useful about the newly created Internet Gateway, I'm not sure we should ignore it